terminus-ui/expansion-panel/src/expansion-panel.component.ts

```diff
@@ -0,0 +1,322 @@
+   * This alias is to conform to our existing naming convention.
+   */
+  @Input()
+  public set isExpanded(value: boolean) {
+    this.expanded = value;
+  }
+  public get isExpanded(): boolean {
+    return this.expanded;
+  }
+
+  /**
+   * The event emitted after the panel body's expansion animation finishes
+   */
+  @Output()
+  readonly afterExpand: EventEmitter<void> = new EventEmitter();
+
+  /**
+   * The event emitted after the panel body's collapse animation finishes
+   */
+  @Output()
+  readonly afterCollapse: EventEmitter<void> = new EventEmitter();
+
+
+  constructor(
+    @Optional() @SkipSelf() @Inject(TS_ACCORDION) accordion: TsAccordionBase,
+    _changeDetectorRef: ChangeDetectorRef,
+    _uniqueSelectionDispatcher: UniqueSelectionDispatcher,
+    private _viewContainerRef: ViewContainerRef,
+    private documentService: TsDocumentService,
+    @Optional() @Inject(ANIMATION_MODULE_TYPE) public animationMode?: string,
+    @Inject(TS_EXPANSION_PANEL_DEFAULT_OPTIONS) @Optional() defaultOptions?: TsExpansionPanelDefaultOptions,
+  ) {
+    super(accordion, _changeDetectorRef, _uniqueSelectionDispatcher);
+
+    this.accordion = accordion;
+    this.document = documentService.document;
+
+    // We need a Subject with distinctUntilChanged, because the `done` event fires twice on some browsers.
+    // See https://github.com/angular/angular/issues/24084
+    this.bodyAnimationDone.pipe(
+      untilComponentDestroyed(this),
+      distinctUntilChanged((x, y) => {
+        return x.fromState === y.fromState && x.toState === y.toState;
+      }),
+    ).subscribe((event) => {
+      // istanbul ignore else
+      if (event.fromState !== 'void') {
+        if (event.toState === 'expanded') {
+          this.afterExpand.emit();
+        } else if (event.toState === 'collapsed') {
+          this.afterCollapse.emit();
+        }
+      }
+    });
+
+    if (defaultOptions) {
+      this.hideToggle = defaultOptions.hideToggle;
+    }
+  }
+
+
+  /**
+   * If a lazy-loaded template exists, inject it after the panel is opened
+   */
+  public ngAfterContentInit(): void {
+    // istanbul ignore else
+    if (this.lazyContent) {
+      // Render the content as soon as the panel becomes open.
+      this.opened.pipe(
+        // tslint:disable: no-non-null-assertion
+        startWith<void>(null!),
+        // tslint:enable: no-non-null-assertion
+        filter(() => this.expanded && !this.portal),
+        take(1),
+      ).subscribe(() => {
+        this.portal = new TemplatePortal(this.lazyContent.template, this._viewContainerRef);
+      });
+    }
+  }
+
+
+  /**
+   * Send any input changes through the Subject stream
+   */
+  public ngOnChanges(changes: SimpleChanges): void {
+    this.inputChanges.next(changes);
+  }
+
+
+  /**
+   * Destroy the parent and finalize any subscriptions
+   */
+  public ngOnDestroy(): void {
+    super.ngOnDestroy();
+    this.inputChanges.complete();
```

Comment:
Would animation has weird behavior if we don't call `this.bodyAnimationDone.complete()`?

Reply:
Not to my knowledge. We're using `untilComponentDestroyed` so it should automatically get completed.

I could probably have added the same pipe operator to `inputChanges` instead of calling complete here, but I just never took the time to go look through the base cdk components to see if anyone else was subscribing to this one.